python/python/lance/dataset.py

```diff
@@ -1988,6 +1989,17 @@ def write_dataset(
         a custom class that defines hooks to be called when each fragment is
         starting to write and finishing writing.
     """
+    try:
+        # Huggingface datasets
+        import datasets
+
+        if isinstance(data_obj, datasets.Dataset):
+            if schema is None:
+                schema = data_obj.features.arrow_schema
```

Comment:
for the datasets that have embeddings, are they usually list or fsl? do we need to check/cast or anything like that?

Reply:
It is too smart at the lance level tho.